### PR TITLE
PHP 8.1 + Symfony 6.0

### PR DESCRIPTION
**Pull request by ruudk**
Fixes:
- #676 
- #677

### CODE REVIEW

**Comment by ruudk** [comment]
@mihaileu  It's ready :) I had to drop the older PHP versions to be able to upgrade to the latest PHPUnit version. Since they are not supported anyway, I think this is good.

**Comment by javer** [comment]
@ruudk Please resolve conflicts, and I hope this PR will be merged soon.

**Comment by ruudk** [comment]
@javer done

**Comment by javer** [comment]
@mihaileu Could you please review this PR?

**Comment by Eloar** [comment]
What's the point exactly? Is there any problem with using this bundle with newer Symfony or PHP version without those changes? So far bundle works with older versions and doesn't force users to upgrade whole projects just because this Bundle updated.

**Comment by ruudk** [comment]
> What's the point exactly? Is there any problem with using this bundle with newer Symfony or PHP version without those changes? So far bundle works with older versions and doesn't force users to upgrade whole projects just because this Bundle updated.

I wouldn't have done this work if it wasn't needed obviously.

> Also it is very good practice to squash commits for PR.

That's your opinion. That's not how I work.

**Comment by javer** [comment]
@Eloar Of course it cannot be used with newer Symfony version, because it cannot even be installed at all:
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - php-amqplib/rabbitmq-bundle[2.10.0, ..., 2.10.2] require symfony/dependency-injection ^4.3|^5.0 -> found symfony/dependency-injection[v4.3.0, ..., v4.4.34, v5.0.0, ..., v5.4.1] but the package is fixed to v6.0.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires php-amqplib/rabbitmq-bundle 2.10.* -> satisfiable by php-amqplib/rabbitmq-bundle[2.10.0, 2.10.1, 2.10.2].
```

That's how composer constraints work:
https://github.com/php-amqplib/RabbitMqBundle/blob/777bd9078dd892a395135f4ca67833042a093045/composer.json#L13-L21

**Review by javer — APPROVED** [review]

**Comment by javer** [comment]
Can confirm that proposed changes works great in the real project with Symfony 6.0 and PHP 8.1.

**Comment by Eloar** [comment]
So there is no point for dropping support, just extending it for new Symfony version. Old Symfony versions support is not a problem, the problem is lack of Symfony6 support.

**Comment by ruudk** [comment]
@Eloar Feel free to create a PR and do it your way.

**Comment by Eloar** [comment]
@ruudk to conflict changes in active Pull Request? That would be counterproductive. Decision is up to Bundle owner I just shared opinion. I like how liberal this bundle is.

**Comment by javer** [comment]
PHPUnit 9.3+ is required to support PHP 8.0, and PHPUnit 9.5 is actually required to support PHP 8.1.
PHPUnit 9.5 requires `php >=7.3`, so we have to drop at least 7.1 and 7.2 which are beyond their EOL for at least one year.

**Comment by Eloar** [comment]
@javer is there a point to drop PHPUnit ^7.5|^8.5 version constraint and not just extend it with ^9.5 for PHP 8.1?

**Comment by mihaileu** [comment]
@ruudk  @javer thx for this PR 
I'll take a peak in this PR. For us it's important to keep the compatibility with the current allowed sy version in order to no create a BC  and increase major. Otherwise we will need to maintain 2 versions of this library. I'll look into to see if we cover this situation

**Comment by ramunasd** [comment]
Regarding PHPUnit compatibility, You can use same workaround as we did here https://github.com/php-amqplib/php-amqplib/commit/393546c5ee0a818a897a6fa8ca05e73aa95a00ef#diff-96a82592013e5f4b91e682332583b7a4a6ca1ff0431d267c61179b1a21b167daR6

**Comment by ruudk** [comment]
Feel free to use my PR and improve it further. I'm not going to change it and support PHP versions that are EOL. Sorry. It took me already way too much time.

File: composer.json [hunk 1]
```diff
@@ -8,23 +8,23 @@
...
     "require": {
-        "php":                          "^7.1|^8.0",
+        "php":                          "^7.4|^8.0",
 
-        "symfony/dependency-injection": "^4.3|^5.0",
-        "symfony/event-dispatcher":     "^4.3|^5.0",
-        "symfony/config":               "^4.3|^5.0",
-        "symfony/yaml":                 "^4.3|^5.0",
-        "symfony/console":              "^4.3|^5.0",
+        "symfony/dependency-injection": "^4.4|^5.3|^6.0",
```
[the remark this answers is not in the document]
Comment (ruudk) [reply composer.json:13]: https://symfony.com/releases

File: DependencyInjection/OldSoundRabbitMqExtension.php [hunk 2]
```diff
@@ -654,7 +654,7 @@ protected function injectConnection(Definition $definition, $connectionName)
         $definition->addArgument(new Reference(sprintf('old_sound_rabbit_mq.connection.%s', $connectionName)));
     }
 
-    public function getAlias()
+    public function getAlias() : string
```
[the remark this answers is not in the document]
Comment (ruudk) [reply DependencyInjection/OldSoundRabbitMqExtension.php:657]: php-cs-fixer should do this.

[the remark this answers is not in the document]
Comment (ramunasd) [reply DependencyInjection/OldSoundRabbitMqExtension.php:657]: Let's add PSR12 style check for all PR's. These things should be noticeable automatically.

File: composer.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mihaileu) [reply composer.json:13]: @ramunasd I see no reason to let old compatibility with php and symfony EOL. We will keep same major tag an increase the minor. There's kinda enough time passed in order to not affect too much people or at lease suggest them an update. What do you think ?

File: DependencyInjection/OldSoundRabbitMqExtension.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (mihaileu) [reply DependencyInjection/OldSoundRabbitMqExtension.php:657]: i'll add php-cs-fixer as github action

File: composer.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ramunasd) [reply composer.json:13]: AFAIK symfony 4.4 and 5.4 are LTS releases, we can stick to them

**Comment by ruudk** [comment]
Didn't expect this to be merged, thank you! @mihaileu 🎉

**Pull request merged**